src/components/vis/Map/index.jsx

```diff
@@ -0,0 +1,235 @@
+
+      union.forEach(parsedLayer => {
+        if (!setOld.has(parsedLayer)) {
+          layer = nextProps.layersActive.filter(l => l.dataset === parsedLayer)[0];
+          this.addLayer(layer);
+        } else if (!setNew.has(parsedLayer)) {
+          layer = _this.props.layersActive.filter(l => l.dataset === parsedLayer)[0];
+          this.removeLayer(layer);
+        } else {
+          // Order
+        }
+      });
+    }
+
+    if (this.props.sidebar.open !== nextProps.sidebar.open) {
+      this.setState({
+        sidebarOpen: nextProps.sidebar.open,
+        sidebarWidth: nextProps.sidebar.width
+      });
+    }
+  }
+
+  shouldComponentUpdate(nextProps, nextState) {
+    const loadingChanged = this.state.loading !== nextState.loading;
+    const sidebarWidthChanged = this.props.sidebar.open !== nextProps.sidebar.open;
+    return loadingChanged || sidebarWidthChanged;
+  }
+
+  componentWillUnmount() {
+    this._mounted = false;
+    // Remember to remove the listeners before removing the map
+    // or they will stay in memory
+    this.props.setMapParams && this.removeMapEventListeners();
+    this.map.remove();
+  }
+
+
+  // SETTERS
+  setLayerManager() {
+    const stopLoading = () => {
+      // Don't execute callback if component has been unmounted
+      this._mounted && this.setState({
+        loading: false
+      });
+    };
+
+    this.layerManager = new this.props.LayerManager(this.map, {
+      onLayerAddedSuccess: stopLoading,
+      onLayerAddedError: stopLoading
+    });
+  }
+
+  setAttribution() {
+    this.map.attributionControl.addAttribution('&copy; <a href="http://www.openstreetmap.org/copyright" target="_blank">OpenStreetMap</a>');
+  }
+
+  setZoomControl() {
+    this.map.zoomControl && this.map.zoomControl.setPosition('topright');
+  }
+
+  setBasemap() {
+    this.tileLayer = L.tileLayer(config.BASEMAP_TILE_URL, {})
+                      .addTo(this.map)
+                      .setZIndex(0);
+  }
+
+  // GETTERS
+  getMapParams() {
+    const params = {
+      zoom: this.getZoom(),
+      latLng: this.getCenter()
+    };
+    return params;
+  }
+
+  // MAP FUNCTIONS
+  getCenter() { return this.map.getCenter(); }
+
+  getZoom() { return this.map.getZoom(); }
+
+  fitBounds(geoJson, sidebarWidth) {
+    const geojsonLayer = L.geoJson(geoJson);
+    this.map.fitBounds(geojsonLayer.getBounds(), {
+      paddingTopLeft: [sidebarWidth || 0, 0],
+      paddingBottomRight: [0, 0]
+    });
+  }
+
+
+  // MAP LISTENERS
+  setMapEventListeners() {
+    function mapChangeHandler() {
+      // Dispatch the action to set the params
+      this.props.setMapParams(this.getMapParams());
+    }
+
+    if (this.props.setMapParams) {
+      this.map.on('zoomend', mapChangeHandler.bind(this));
+      this.map.on('dragend', mapChangeHandler.bind(this));
+    }
+  }
+
+  removeMapEventListeners() {
+    this.map.off('zoomend');
+    this.map.off('dragend');
+  }
+
+  // LAYER METHODS
+  addLayer(layer, filters) {
```

Comment:
We can join these two functions `addLayer` and `addLayers` into one that checks if the layers passed are an array or an object. The same with `removeLayer` and `removeLayers`